src/main/scala/com/microsoft/hyperspace/actions/RefreshAction.scala

```diff
@@ -42,12 +42,11 @@ class RefreshAction(
 
   private lazy val previousIndexLogEntry = previousLogEntry.asInstanceOf[IndexLogEntry]
 
-  // Deserialize the plan and create a df.
+  // Reconstruct a df from schema
   private lazy val df = {
-    val serializedPlan = previousIndexLogEntry.source.plan.properties.rawPlan
-    val plan = LogicalPlanSerDeUtils.deserialize(serializedPlan, spark)
-    val qe = spark.sessionState.executePlan(plan)
-    new Dataset[Row](spark, plan, RowEncoder(qe.analyzed.schema))
+    val rel = previousIndexLogEntry.relation
+    val dataSchema = DataType.fromJson(rel.dataSchemaJson).asInstanceOf[StructType]
+    spark.read.schema(dataSchema).load(rel.rootPaths: _*)
```

Comment:
seems like default data source is parquet ([link](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala#L898)).
Could you add file format as well to this statement (to support csv etc.)

Reply:
Thanks for pointing out this! I updated the pr and added a test for it.